### DISCUSSION TITLE
Ask questions about data storage

### DISCUSSION THREAD

**Discussion by AbelHugh** [prometheus-operator/kube-prometheus#1386] [Q&A] [answered]
I used the 0.8.0 version, my kubernetes version is 1.20, and it runs normally, but I would like to ask, if my monitoring data is to be retained for 30 days, how should I set it in the yaml file? In addition, if I want to How to configure pvc if the data is retained

**Comment by paulfantom** [comment, marked as answer]
For retention you need to configure `retention` setting in `prometheus.prometheus.spec.retention` - example https://github.com/thaum-xyz/ankhmorpork/blob/master/apps/monitoring/jsonnet/main.jsonnet#L234

For pvc you need to configure `storage` setting in `prometheus.prometheus.spec.storage` - example https://github.com/thaum-xyz/ankhmorpork/blob/master/apps/monitoring/jsonnet/main.jsonnet#L242-L255
